src/firefly/js/ui/panel/CollapsiblePanel.jsx

```diff
@@ -25,85 +24,87 @@ export const CollapseHeaderCorner = {
                             BottomLeft: 0b1000
                         };
 
-function collapsibleStateFromProps(props) {
-    // component key needs to be defined if the state needs to be saved though unmount/mount
-    let isOpen = props.componentKey && getComponentState(props.componentKey).isOpen;
-    if (!isBoolean(isOpen)) {
-        // use property to initialize state
-        isOpen = props.isOpen? props.isOpen: false;
-    }
-
-    return {isOpen};
-}
 
-export class CollapsiblePanel extends PureComponent {
 
-    constructor(props) {
-        super(props);
-
-        this.state = collapsibleStateFromProps(props);
-
-        this.handleClick = this.handleClick.bind(this);
-        this.getContentHeight = this.getContentHeight.bind(this);
+export function CollapsiblePanel({
+    componentKey,
+    header,
+    isOpen=false,
+    headerRoundCorner=0,
+    headerStyle,
+    contentStyle,
+    wrapperStyle,
+    borderStyle=CollapseBorder.Oneborder,
+    onToggle,
+    children
+}) {
+    function getIsOpenFromGlobalState() {
+        // component key needs to be defined if the state needs to be saved though unmount/mount
+        let isOpen_ = componentKey && getComponentState(componentKey).isOpen;
+        if (!isBoolean(isOpen_)) {
+            // use property to initialize state
+            isOpen_ = isOpen? isOpen: false;
+        }
+        return isOpen_;
     }
+    const isOpenGlobalState = getIsOpenFromGlobalState();
 
-    static getDerivedStateFromProps(props) {
-        return collapsibleStateFromProps(props);
-    }
+    const [isOpenState, setIsOpenState] = useState(isOpenGlobalState);
```

Comment:
Yes I thought of doing that but we also need this in L55 (`if (isOpenGlobalState !== isOpenState) setIsOpenState(isOpenGlobalState);`), that's why I stored it in a variable so that it can be reused.

Reply:
ok let me look that line.